app/models/user.rb

```diff
@@ -40,6 +41,10 @@ def forget
     update remember_digest: nil
   end
 
+  def current_user? user
+    self == user
```

Comment:
how many similar method like `==`, what is difference ?

Reply:
có 2 cách so sánh anh ạ . Có thêm cả === và nó khác nhau ở đây phụ thuộc vào các lớp mà ta cài đặt. 

Range | Trả về true nếu đối tượng là 1 thành phần của range, false trong trường hợp khác
Regexp | So sánh 2 regexp như là chuỗi
Module | Trả về true nếu đối tượng là thể hiện của module và thể hiện của module con

Reply:
===, equal?, eql? nhé :v

Reply:
Method eql? được sử dụng để kiểm tra giá trị của các phần tử trong một hàm
Chúng ta có một class , giá trị truyền vào khi khởi tạo sẽ được gán cho cho biến . Tiếp theo chúng ta thực hiện ghi đè phương thức epl?
Và method này trả về True nếu như hai object có biến cùng kiểu

equal? sẽ trả về kết quả là true chỉ khi nó (đối tượng bị gọi bởi equal?) và tham số (đối tượng được gọi bởi equal?) là cùng một đối tượng
=> khi cả hai obj cùng trỏ đến 1 obj
Và nó ko thể ghi đè